api/v1/ytsaurus_types.go

```diff
@@ -479,11 +479,12 @@ type UISpec struct {
 	//+kubebuilder:default:=NodePort
 	ServiceType  corev1.ServiceType `json:"serviceType,omitempty"`
 	HttpNodePort *int32             `json:"httpNodePort,omitempty"`
-	//+kubebuilder:default:=true
+	// Use secure connection to the cluster's http-proxies.
+	//+kubebuilder:default:=false
 	//+optional
-	UseInsecureCookies bool                        `json:"useInsecureCookies"`
```

Comment:
Let's not delete but deprecate the field, giving users the opportunity to migrate.
I suggest we add validation that values don't conflict (smth like `UseInsecureCookies == !Secure`) and after that use `Secure` field in the code as you did it.

Reply:
Yeah. This seems reasonable way of deperecation.
Remove default=true and convert into *bool - to not trigger validation if this field isn't here.

I hope validation is called before each reconcile. If not we should do so.

Reply:
`UseInsecureCookies = false` did not make any sense before this change. So I added check for ambiguous `UseInsecureCookies == false && Secure == false`, ignoring `UseInsecureCookies == true`.

Reply:
> UseInsecureCookies = false did not make any sense before this change
I don't get why it didn't make sense. useInsecureCookies was true by default and `YT_AUTH_ALLOW_INSECURE` was set in this case, but if user wanted to disallow using insecure cookies — he might set false and expect UI to use secure cookie.

So now we are making this nil+optional and should think about 3 cases:
1) users who didn't have useInsecureCookies set in their spec — default behaviour with insecure cookie should work for them (which is work since secure is false by default)
2) users who explicitely set useInsecureCookies=true in their spec — they should be fine since we check secure which is false by default and ignore the old field.
3) users who explicitly set useInsecureCookies=false — their spec will be broken after operator update, so we must defend them and add validation (which is what you did).

So I think current code shouldn't break anyone.